### PR TITLE
nicolas/dash 329 dapi delete ability to delete passkey

### PR DESCRIPTION
**Pull request by NicolasLopes7**
This PR implements a new method from a new BAPI endpoint to delete passkeys. It's all under the users' client.

### CODE REVIEW

File: user/api.go [hunk 1]
```diff
@@ -90,6 +90,11 @@ func ListOrganizationMemberships(ctx context.Context, id string, params *ListOrg
 	return getClient().ListOrganizationMemberships(ctx, id, params)
 }
 
+// DeletePasskey deletes a passkey by it's identification ID.
```
Comment (thiskevinwang) [inline user/api.go]:
```suggestion
// DeletePasskey deletes a passkey by its identification ID.
```

File: user/client.go [hunk 2]
```diff
@@ -428,3 +428,15 @@ func (c *Client) ListOrganizationMemberships(ctx context.Context, id string, par
 	err = c.Backend.Call(ctx, req, list)
 	return list, err
 }
+
+// DeletePasskey deletes a passkey by it's identification ID.
```
Comment (thiskevinwang) [inline user/client.go]:
```suggestion
// DeletePasskey deletes a passkey by its identification ID.
```

File: user/client_test.go [hunk 3]
```diff
@@ -422,3 +422,22 @@ func TestUserClientListOrganizationMemberships(t *testing.T) {
...
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T:      t,
+			Out:    json.RawMessage(fmt.Sprintf(`{"id":"%s"}`, passkeyIdentificationID)),
+			Method: http.MethodDelete,
+			Path:   "/v1/users/" + userID + "/passkeys/" + passkeyIdentificationID,
+		},
+	}
+	client := NewClient(config)
+	passkey, err := client.DeletePasskey(context.Background(), userID, passkeyIdentificationID)
```
Comment (thiskevinwang) [inline user/client_test.go:440]: ❓ Do we need to seed a passkey?
What happens if we call `client.DeletePasskey` twice in this test?

Comment (NicolasLopes7) [reply user/client_test.go:440]: the test only asserts the http call for BAPI through the transporter. Defined [here](https://github.com/clerk/clerk-sdk-go/blob/3445b5e9a20ceca1f1317c70bf21cbcf81667dee/user/client_test.go#L432-L437)

Comment (thiskevinwang) [reply user/client_test.go:440]: I see, so `Out` is effectively the mocked-expected-response

**Review by thiskevinwang — APPROVED** [review]

**Review by mzhong9723 — APPROVED** [review]

**Pull request merged**